### PR TITLE
Update TransportXPackInfoAction to only query features that are present

### PR DESCRIPTION
**Pull request by ChrisHegarty**
This commit updates the TransportXPackInfoAction to only query features that are present, i.e. features that have a monitor info action registered. This allows for some features to not be present, e.g. ES|QL, and maybe others in the future.

### CODE REVIEW

File: x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/action/TransportXPackInfoActionTests.java [hunk 1]
```diff
@@ -93,6 +83,100 @@ protected List<XPackInfoFeatureAction> infoActions() {
         categories.add(XPackInfoRequest.Category.FEATURES);
         request.setCategories(categories);
 
+        XPackInfoResponse response = getResponse(request, client, licenseService, new ArrayList<>(featureSets.keySet()));
+
+        if (request.getCategories().contains(XPackInfoRequest.Category.BUILD)) {
```
Comment (ChrisHegarty) [inline x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/action/TransportXPackInfoActionTests.java:88]: Not sure why GH renders the diffs this way, these are not new assertions. The existing test has just a little refactoring to allow to better share across the new test scenario.

**Comment by ChrisHegarty** [comment]
This is useful for deployments of Elasticsearch where some modules are not present in the distribution, e.g. for Serverless.

**Comment by jakelandis** [comment]
The code level change looks fine, but this has downstream implications in reading the response.  IIUC the response today is a fixed list that it will always return available or enabled for each feature. I believe that Kibana calls this API for telemetry and is might also be used for other business intelligence (and possibly billing ?) so I wonder if having variable responses would cause those downstream systems issues.  I also assume this is broken in serverless due to calling actions that don't exist there ?

I wonder if a different fix would be to keep the fixed return list for all possible features, but don't call underlying action at all if not found in the client and for that case fake the response such that is neither available or enabled. 

(also, I am not an owner of this area of code but happy help)

**Comment by ChrisHegarty** [comment]
That's a good idea @jakelandis.  While I think that it would be generally useful to support usage and info requests in this way, I'm going to withdraw this change of now, in favour of adding targeted specific dummy actions that return unavailable, false, for particular actions, as is done in serverless, e.g. for ILM.

**Pull request closed without merging**